src/Agent/NewRelic/Profiler/Configuration/Configuration.h

```diff
@@ -630,25 +632,34 @@ namespace NewRelic { namespace Profiler { namespace Configuration {
         /// <summary>
         /// Returns 0 if the process should not be instrumented, 1 if it should be instrumented, and -1 if it is indeterminate.
         /// </summary>
-        int ShouldInstrumentAzureFunction(xstring_t const& appPoolId, xstring_t const& commandLine)
+        int ShouldInstrumentAzureFunction(xstring_t const& processPath, xstring_t const& appPoolId, xstring_t const& commandLine)
         {
             LogInfo(_X("Azure function detected. Determining whether to instrument ") + commandLine);
 
-            bool isAzureWebJobsScriptWebHost = NewRelic::Profiler::Strings::ContainsCaseInsensitive(commandLine, appPoolId);
+            bool isAzureWebJobsScriptWebHost = appPoolId.length() > 0  && NewRelic::Profiler::Strings::ContainsCaseInsensitive(commandLine, appPoolId);
             if (isAzureWebJobsScriptWebHost)
             {
                 LogInfo(L"Appears to be Azure WebJobs Script WebHost based on commandLine. Not instrumenting this process.");
                 return 0;
             }
 
             // AzureFunctionsNetHost.exe is the typical startup command for Azure Functions
-            bool isAzureFunctionsNetHost = NewRelic::Profiler::Strings::ContainsCaseInsensitive(commandLine, _X("FunctionsNetHost.exe"));
+             
+            bool isAzureFunctionsNetHost = Strings::EndsWith(processPath, _X("FUNCTIONSNETHOST.EXE"));
             if (isAzureFunctionsNetHost)
             {
                 LogInfo(L"FunctionNetHost.exe is a valid Azure function command. This process will be instrumented.");
                 return 1;
             }
 
+            // func.exe is the local test tool and should not be instrumented
+            bool isFuncExe = Strings::EndsWith(processPath, _X("FUNC.EXE"));
```

Comment:
Do we think this check is specific enough?

Reply:
I think so, it'll only be hit if the `FUNCTIONS_WORKER_RUNTIME` environment variable is present, so we know at this point that we're in Azure function mode.